netket/_vmc.py

```diff
@@ -158,7 +160,7 @@ def _forward_and_backward(self):
 
             dp = _np.empty(self._npar, dtype=_np.complex128)
 
-            self._sr.compute_update(_der_logs, grad, dp)
+            dp = self._sr.compute_update(_der_logs, grad, dp)
```

Comment:
No need to assign `dp` here, its the same array that is passed in as `out` argument already.

Reply:
Yes for some reason that I don't quite understand I need to do the explicit assignment otherwise dp is not modified... still looking into this

Reply:
Ah, that's because `out` is reassigned instead of written to in line 123 (and equivalent lines further below).

Probably it should be `out[:], ... = self._sparse_solver(...)`.

Reply:
yeah thanks!